### DISCUSSION TITLE
Can i run a nodejs function on demand when showing a story?

### DISCUSSION THREAD

**Discussion by thebspin** [storybookjs/storybook#26250] [Help]
### Summary

I would like to show some content on a page that is currently coming from a prebuild json, but i rather see it coming from the script that builds it at runtime in Storybook itself. Because prebuilding the json can be forgotten everytime. This node script would read local files from the repo. 

Important would be that it only does this when serving SB in dev mode, when it is being build in the pipeline and deployed it's fine that it gets it context from a json (that is being build in the pipeline)

**Comment by Marklb** [comment]
What I think you are asking is something that could probably be implemented as a plugin to whatever builder you are using. I don't think there is a simple generic answer, at least with the limited info provided.

If you have some process that is periodically changing a file then the builder could detect and rebuild with the new file, but I don't think that is what you are asking.

If you need the script to run, without triggering a rebuild then maybe a partial rebuild, similar to how Hot Module Reload works.

To break it down to a more basic concept, that isn't specific to the build tool, you could think of it as an api. When you start your dev build, you could open a websocket *or any other communication from a browser* and have your story request the info to be retrieved and returned.

With more specific info provided, such as the framework and build tool, someone may have a better answer, but I don't think there is really a simple generic answer.
